src/util/std_types.h

```diff
@@ -1402,6 +1403,17 @@ inline pointer_typet &to_pointer_type(typet &type)
   return static_cast<pointer_typet &>(type);
 }
 
+template<> inline bool can_cast_type<pointer_typet>(const typet &type)
+{
+  return type.id()==ID_pointer;
+}
+
+inline void validate_type(const pointer_typet &type)
+{
+  DATA_INVARIANT(!type.get(ID_width).empty(), "pointer must have width");
+  DATA_INVARIANT(type.get_width() > 0, "pointer must have non-zero width");
```

Comment:
To check that these really are invariants (I believe they should be but it is always good to check), could we add a call to this validate_type to to_pointer_type, that was we aren't enforcing different invariants depending on which system you use.